irods_audit_elk_stack/README.md

```diff
@@ -0,0 +1,136 @@
+
+[^2]: At time of writing, the audit plugin still inserts the `__BEGIN_JSON__` and `__END_JSON__` tokens.
+
+## Updating This Container
+
+### General Updates
+
+Other than the Ubuntu release itself and the major versions for Temurin, Elasticsearch, and Kibana, the Dockerfile does not specify specific versions of software to be used; therefore, packages can be updated to their latest versions by simply rebuilding the docker image without cache.
+
+### Updating to a new Ubuntu release
+
+Assuming the desired version of Ubuntu introduces no breaking changes and is supported by the third-party apt repositories, the release can be changed in the `FROM` instruction in the Dockerfile.
+
+When updating the Dockerfile, please only use LTS Ubuntu releases, as non-LTS releases are unlikely to have support from third-party apt repositories.
+
+### Updating Java to a new major release
+
+The flavor and major version of Java can be specified at image build-time with the `java_ver`, `java_vendor`, and `java_dist` arguments. (See the Dockerfile for more info.)
+
+When updating the Dockerfile, please only use LTS Java versions, as minor ElasticSearch releases will sometimes drop support for older non-LTS Java versions.
+
+### Updating Elasticsearch and Kibana to a new major release
+
+The major version of Elasticsearch and Kibana can be specified at image build-time with the `es_ver` argument. (See the Dockerfile for more info.)
+
+Elasticsearch is picky about the Java versions it will run on, so changing the Elasticsearch version will often necessitate a change to the Java version as well. Consult the [official Elasticsearch support matrix](https://www.elastic.co/support/matrix#matrix_jvm) for more information.
+
+It is very possible for configuration schema, environment variables, command line arguments, and API endpoints to change between major releases of Elasticsearch and Kibana. Consult [official](https://www.elastic.co/guide/en/elasticsearch/reference/current/rest-apis.html) [documentation](https://www.elastic.co/guide/en/kibana/current/api.html), default [configuration](https://github.com/elastic/elasticsearch/tree/main/distribution/src/config) [files](https://github.com/elastic/kibana/tree/main/config), systemd [unit](https://github.com/elastic/elasticsearch/blob/main/distribution/packages/src/common/systemd/elasticsearch.service) [files](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/service_templates/systemd/usr/lib/systemd/system/kibana.service), and default [environment](https://github.com/elastic/elasticsearch/blob/main/distribution/packages/src/common/env/elasticsearch) [files](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/service_templates/env/kibana) when updating to a new major release of Elasticsearch and Kibana.
```

Comment:
I read through the rendered version of this first.

Things I noticed regarding adjacent hyperlinks (e.g. `default [configuration](link) [files](link)`):
- They can confuse the reader
    - I found myself wondering why both words weren't being underlined when I moused over one
- The link names didn't help me understand what information would be presented upon clicking them
    - After a minute or two, I realized that one link was for elastic while the other was for kibana

Consider changing this so that the reader knows what to expect. That could be through the use of more words, a list, etc.

Reply:
Should be more clear now